opentelemetry/src/context.rs

```diff
@@ -381,6 +389,75 @@ impl Hasher for IdHasher {
     }
 }
 
+struct ContextStack {
+    current_cx: Context,
+    current_id: usize,
+    // TODO:ban wrap the whole id thing in its own type
+    id_count: usize,
+    // TODO:ban wrap the the tuple in its own type
+    stack: Vec<Option<(usize, Context)>>,
+}
+
+impl ContextStack {
+    #[inline(always)]
+    fn push(&mut self, cx: Context) -> usize {
+        self.id_count += 512; // TODO:ban clean up this
```

Comment:
Is this to make relying on the id value opaque?
It limits the context stack to 511 nested contexts, which is not that terrible since it's already a huge stack trace.
 But  I'm afraid that if we reach this number we will start looping over. So the 512th context will be mapped to id 0, so we won't do anything on popping so we won't change the current context and context stack will never be freed.
 
 Maybe an opaque wrapper is better?

Reply:
Indeed an opaque wrapper is better, and as the description of the PR mentioned, the edge cases are ignored until ppl are ok with the general approach. I'm actually not sure if there needs to be any other `id` part to the `ContextStack` than the array position. I mean how would you mess up the `ContextGuard` since it's `!Send` and doesn't implement `Copy` or `Clone`.